### PR TITLE
Allow uvflag to read posixpaths.

### LINKED ISSUE #542: update docstrings to put parameter and return types in the right place
To ensure that they're rendered properly in the docs.

example:
Args:
     param1 (int):
           description of parameter 1
     param2 (list(int)):
           description of parameter 2

### LINKED ISSUE #777: rename `input` in UVFlag.__init__
because it is shadowing a python builtin

### LINKED ISSUE #870: Let UVFlag take PosixPath as input
https://github.com/RadioAstronomySoftwareGroup/pyuvdata/blob/16eeca3edca53a77119f851730c79b187fd253cb/pyuvdata/uvflag/uvflag.py#L466

e.g.
```
from pyuvdata import UVFlag
import pathlib

myfile = pathlib.Path('mypath')
uvf = UVFlag(myfile)
```
Will result in an error because it doesn't know how to handle PosixPath objects

### PR DESCRIPTION
**Pull request by adampbeardsley**
## Description

Added `pathlib.PosixPath` to allowable types passed to `UVFlag.__init__`.
Also updated coverage script to build extensions.
Renamed input -> indata in UVFlag to avoid shadowing a python builtin.

## Motivation and Context

`PosixPath`s are awesome, and we should be able to pass them into UVFlag. Already works for UVData and UVCal, but UVFlag has an explicit type check that did not include `PosixPath`. Functionally everything is the same.
Closes #870 
Closes #777
Closes #542

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation change (documentation changes only)
- [ ] Version change
- [ ] Build or continuous integration change

## Checklist:

- [x] I have read the [contribution guide](https://github.com/RadioAstronomySoftwareGroup/pyuvdata/blob/master/.github/CONTRIBUTING.md).
- [x] My code follows the code style of this project.

New feature checklist:
- [x] I have added or updated the docstrings associated with my feature using the [numpy docstring format](https://numpydoc.readthedocs.io/en/latest/format.html).
- [ ] I have updated the tutorial to highlight my new feature (if appropriate).
- [x] I have added tests to cover my new feature.
- [x] All new and existing tests pass.
- [x] I have updated the [CHANGELOG](https://github.com/RadioAstronomySoftwareGroup/pyuvdata/blob/master/CHANGELOG.md).

### CODE REVIEW

**Comment by adampbeardsley** [comment]
Changed to using `pathlib.Path` to work with Windows.
Also changed `input` variable to `indata` to fix shadowing.
Closes #777

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,7 @@ All notable changes to this project will be documented in this file.
 ## [Unreleased]
 
 ### Added
+- `UVFlag` can now take `PosixPath` objects as input.
```
Comment (bhazelton) [inline CHANGELOG.md:7]: should we also call out the `input` rename in the changelog?

File: scripts/test_coverage.sh [hunk 2]
```diff
@@ -3,7 +3,7 @@
 DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 cd $DIR/..
 
-pip install .
+pip install -e .
```
Comment (mkolopanis) [inline scripts/test_coverage.sh:6]: This addition basically turns this script into the `test_cover_no_install.sh` script. :thinking: Why would it not work with just `pip install .`?

File: scripts/test_coverage.sh [hunk 3]
```diff
@@ -3,7 +3,7 @@
 DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 cd $DIR/..
 
-pip install .
+pip install .[test]
```
Comment (mkolopanis) [inline scripts/test_coverage.sh:6]: This uh _could_ cause trouble for users with zsh as their shell. But I think it'll probably be fine. Those who that do use it probably can make workarounds

**Pull request merged**